src/components/GridLattice.vue

```diff
@@ -12,7 +12,16 @@ const props = defineProps<{
   labels: string[]
   colors: string[]
   heldNotes: Set<number>
+  baseMidiNote:number
+  onTouchStart:TouchEventCallback
+  onTouchEnd:TouchEventCallback
+  onMouseDown:MouseEventCallback
+  onMouseUp:MouseEventCallback
+  onMouseEnter:MouseEventCallback
+  onMouseLeave:MouseEventCallback
 }>()
+type TouchEventCallback = (event: TouchEvent, index: number) => void
```

Comment:
These types are repeated multiple times. Make a `src/types.ts` or similar so that they can be tweaked from a single location in the future if needed.